### PR TITLE
Added list of scopes in config

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDirectives.scala [hunk 1]
```diff
@@ -24,7 +24,9 @@ trait OAuthDirectives {
     }
 
   def hasOneOfTheScopes(authorizedUser: AuthorizedUser)(scope: Scope*): Directive0 = {
-    authorizedUser.scope.toSet.intersect(scope.toSet).isEmpty match {
+    println(s"scope: ${authorizedUser.scope}")
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDirectives.scala]: `println` is nice debugging, but it should not be merged to master

Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDirectives.scala]: fixed

**Comment by aryszka** [comment]
nice!

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDomain.scala [hunk 2]
```diff
@@ -2,18 +2,20 @@ package org.zalando.spearheads.innkeeper.oauth
...
 import com.google.inject.{ Inject, Singleton }
 import com.typesafe.config.Config
+import scala.collection.immutable.List
+import scala.collection.JavaConversions._
 
 /**
  * @author dpersa
  */
-case class Scope(val name: String) extends AnyVal
+case class Scope(val scopeNames: List[String]) extends AnyVal
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDomain.scala]: Why `List`? Seems to me that a `Set` is a better fit. Or at least `Seq`. But List is overly specific

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthJsonProtocol.scala [hunk 3]
```diff
@@ -9,11 +9,11 @@ object OAuthJsonProtocol extends DefaultJsonProtocol {
 
   implicit object ScopeFormat extends RootJsonFormat[Scope] {
 
-    override def write(scope: Scope) = JsString(scope.name)
+    override def write(scope: Scope) = JsArray(scope.scopeNames.map(scopeName => JsString(scopeName)).toVector)
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthJsonProtocol.scala]: is `toVector` really necessary? Anyway, `scope.scopeNames.map(JsString(_))` is a bit less verbose

Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthJsonProtocol.scala]: the api expects a vector :( not sure why

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala [hunk 4]
```diff
@@ -40,7 +43,13 @@ class OAuthService @Inject() (val config: Config,
...
       val json = Await.result(futureJson, 1.second)
       json.parseJson.convertTo[AuthorizedUser]
-    }.toOption
+    } match {
+      case Success(value) => Some(value)
+      case Failure(ex) => {
+        logger.warn("The OAuth exception is", ex)
+        None
+      }
+    }
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: can be rewritten as:
`.transform(identity, {case e => logger.warn("The OAuth exception is", ex); Failure(e)}).toOption`

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthJsonProtocol.scala (hunk 3 above)
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthJsonProtocol.scala]: weird api

**Comment by mo-gr** [comment]
lgtm

**Pull request merged**